doc/task-hierarchy.md

```diff
@@ -8,6 +8,8 @@
   - `Answer Generation -> Fill in the Blank`
   - `Answer Generation -> Multiple Choice Question Answering`
   - `Answer Generation -> Open Question Answering`
+  - `Answer Generation -> Extractive`
+  - `Answer Generation -> Abstractive`
```

Comment:
Since "extractive" and "abstractive" are only defined in the context of "contextual qa", it might make sense to make them nested: 
```
  - `Answer Generation -> Contextual Question Answering`
  - `Answer Generation -> Contextual Question Answering -> Extractive`
  - `Answer Generation -> Contextual Question Answering -> Abstractive`
```

What do you all think? @ghlai9665 @swarooprm